src/Functions/FunctionsCodingIP.cpp

```diff
@@ -514,7 +514,11 @@ class FunctionIPv4ToIPv6 : public IFunction
     static void mapIPv4ToIPv6(UInt32 in, UInt8 * buf)
     {
         unalignedStore<UInt64>(buf, 0);
-        unalignedStore<UInt64>(buf + 8, 0x00000000FFFF0000ull | (static_cast<UInt64>(ntohl(in)) << 32));
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
Use `if constexpr (std::endian::native == std::endian::little)` instead of a macro?